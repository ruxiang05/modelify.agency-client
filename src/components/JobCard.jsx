/* Uses methods and/or components from react, react-router-dom and prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import '../styles/job-card.scss';
import { ReactComponent as DateIcon } from '../assets/icons/calendar-day.svg';
import { ReactComponent as PoundIcon } from '../assets/icons/pound-sign.svg';
import { ReactComponent as LocationIcon } from '../assets/icons/map-marker-alt.svg';
import { ReactComponent as NoteIcon } from '../assets/icons/sticky-note.svg';

const JobCard = ({ job }) => {
  const {
    date, title, address, description, status, _id, pay,
  } = job;
  const formatedDate = new Date(date).toLocaleDateString();
  return (
    <Link
      to={{
        pathname: `/jobs/${_id}`,
        state: job,
      }}
    >
      <div className="job-card">
        <div className="job-card-header">
          <div>
            <DateIcon />
            <p>{formatedDate}</p>
          </div>
          <div>
            <PoundIcon />
            <p>{pay}</p>
          </div>
        </div>
        <div className="job-card-content">
          <h2>{title}</h2>
          <div>
            <LocationIcon />
            <p className="truncate">{address}</p>
          </div>
          <div>
            <NoteIcon />
            <p className="truncate">{description}</p>
          </div>
        </div>
        <div className="job-card-footer">
          <p>{status.toUpperCase()}</p>
        </div>
      </div>
    </Link>
  );
};

JobCard.propTypes = {
  job: PropTypes.shape({}).isRequired,
};

export default JobCard;
