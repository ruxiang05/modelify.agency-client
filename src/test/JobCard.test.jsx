import React from 'react';
import { shallow } from 'enzyme';
import JobCard from '../components/JobCard';

describe('JobCard', () => {
  const job = {
    status: 'in progress',
    _id: '5c97b9cd0328e6b2aa76a361',
    model: '5c686eb24cb1b33456b0be59',
    agent: '5c686e904cb1b33456b0be58',
    title: 'Test',
    date: '2019-05-26T00:00:00.000Z',
    address: 'address',
    pay: 100,
    description: 'some description',
  };
  it('should render JobCard', () => {
    const wrapper = shallow(<JobCard job={job} />);
    expect(wrapper).toMatchSnapshot();
  });
});
