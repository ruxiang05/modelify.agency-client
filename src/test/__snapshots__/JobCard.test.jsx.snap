// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobCard should render JobCard 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <JobCard
    job={
      Object {
        "_id": "5c97b9cd0328e6b2aa76a361",
        "address": "address",
        "agent": "5c686e904cb1b33456b0be58",
        "date": "2019-05-26T00:00:00.000Z",
        "description": "some description",
        "model": "5c686eb24cb1b33456b0be59",
        "pay": 100,
        "status": "in progress",
        "title": "Test",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="job-card"
      >
        <div
          className="job-card-header"
        >
          <div>
            <ForwardRef />
            <p>
              5/26/2019
            </p>
          </div>
          <div>
            <ForwardRef />
            <p>
              100
            </p>
          </div>
        </div>
        <div
          className="job-card-content"
        >
          <h2>
            Test
          </h2>
          <div>
            <ForwardRef />
            <p
              className="truncate"
            >
              address
            </p>
          </div>
          <div>
            <ForwardRef />
            <p
              className="truncate"
            >
              some description
            </p>
          </div>
        </div>
        <div
          className="job-card-footer"
        >
          <p>
            IN PROGRESS
          </p>
        </div>
      </div>,
      "replace": false,
      "to": Object {
        "pathname": "/jobs/5c97b9cd0328e6b2aa76a361",
        "state": Object {
          "_id": "5c97b9cd0328e6b2aa76a361",
          "address": "address",
          "agent": "5c686e904cb1b33456b0be58",
          "date": "2019-05-26T00:00:00.000Z",
          "description": "some description",
          "model": "5c686eb24cb1b33456b0be59",
          "pay": 100,
          "status": "in progress",
          "title": "Test",
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="job-card-header"
          >
            <div>
              <ForwardRef />
              <p>
                5/26/2019
              </p>
            </div>
            <div>
              <ForwardRef />
              <p>
                100
              </p>
            </div>
          </div>,
          <div
            className="job-card-content"
          >
            <h2>
              Test
            </h2>
            <div>
              <ForwardRef />
              <p
                className="truncate"
              >
                address
              </p>
            </div>
            <div>
              <ForwardRef />
              <p
                className="truncate"
              >
                some description
              </p>
            </div>
          </div>,
          <div
            className="job-card-footer"
          >
            <p>
              IN PROGRESS
            </p>
          </div>,
        ],
        "className": "job-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <ForwardRef />
                <p>
                  5/26/2019
                </p>
              </div>,
              <div>
                <ForwardRef />
                <p>
                  100
                </p>
              </div>,
            ],
            "className": "job-card-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <ForwardRef />,
                  <p>
                    5/26/2019
                  </p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "5/26/2019",
                  },
                  "ref": null,
                  "rendered": "5/26/2019",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <ForwardRef />,
                  <p>
                    100
                  </p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 100,
                  },
                  "ref": null,
                  "rendered": 100,
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Test
              </h2>,
              <div>
                <ForwardRef />
                <p
                  className="truncate"
                >
                  address
                </p>
              </div>,
              <div>
                <ForwardRef />
                <p
                  className="truncate"
                >
                  some description
                </p>
              </div>,
            ],
            "className": "job-card-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Test",
              },
              "ref": null,
              "rendered": "Test",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <ForwardRef />,
                  <p
                    className="truncate"
                  >
                    address
                  </p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "address",
                    "className": "truncate",
                  },
                  "ref": null,
                  "rendered": "address",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <ForwardRef />,
                  <p
                    className="truncate"
                  >
                    some description
                  </p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "some description",
                    "className": "truncate",
                  },
                  "ref": null,
                  "rendered": "some description",
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              IN PROGRESS
            </p>,
            "className": "job-card-footer",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "IN PROGRESS",
            },
            "ref": null,
            "rendered": "IN PROGRESS",
            "type": "p",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="job-card"
        >
          <div
            className="job-card-header"
          >
            <div>
              <ForwardRef />
              <p>
                5/26/2019
              </p>
            </div>
            <div>
              <ForwardRef />
              <p>
                100
              </p>
            </div>
          </div>
          <div
            className="job-card-content"
          >
            <h2>
              Test
            </h2>
            <div>
              <ForwardRef />
              <p
                className="truncate"
              >
                address
              </p>
            </div>
            <div>
              <ForwardRef />
              <p
                className="truncate"
              >
                some description
              </p>
            </div>
          </div>
          <div
            className="job-card-footer"
          >
            <p>
              IN PROGRESS
            </p>
          </div>
        </div>,
        "replace": false,
        "to": Object {
          "pathname": "/jobs/5c97b9cd0328e6b2aa76a361",
          "state": Object {
            "_id": "5c97b9cd0328e6b2aa76a361",
            "address": "address",
            "agent": "5c686e904cb1b33456b0be58",
            "date": "2019-05-26T00:00:00.000Z",
            "description": "some description",
            "model": "5c686eb24cb1b33456b0be59",
            "pay": 100,
            "status": "in progress",
            "title": "Test",
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="job-card-header"
            >
              <div>
                <ForwardRef />
                <p>
                  5/26/2019
                </p>
              </div>
              <div>
                <ForwardRef />
                <p>
                  100
                </p>
              </div>
            </div>,
            <div
              className="job-card-content"
            >
              <h2>
                Test
              </h2>
              <div>
                <ForwardRef />
                <p
                  className="truncate"
                >
                  address
                </p>
              </div>
              <div>
                <ForwardRef />
                <p
                  className="truncate"
                >
                  some description
                </p>
              </div>
            </div>,
            <div
              className="job-card-footer"
            >
              <p>
                IN PROGRESS
              </p>
            </div>,
          ],
          "className": "job-card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  <ForwardRef />
                  <p>
                    5/26/2019
                  </p>
                </div>,
                <div>
                  <ForwardRef />
                  <p>
                    100
                  </p>
                </div>,
              ],
              "className": "job-card-header",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <ForwardRef />,
                    <p>
                      5/26/2019
                    </p>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "5/26/2019",
                    },
                    "ref": null,
                    "rendered": "5/26/2019",
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <ForwardRef />,
                    <p>
                      100
                    </p>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": 100,
                    },
                    "ref": null,
                    "rendered": 100,
                    "type": "p",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  Test
                </h2>,
                <div>
                  <ForwardRef />
                  <p
                    className="truncate"
                  >
                    address
                  </p>
                </div>,
                <div>
                  <ForwardRef />
                  <p
                    className="truncate"
                  >
                    some description
                  </p>
                </div>,
              ],
              "className": "job-card-content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Test",
                },
                "ref": null,
                "rendered": "Test",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <ForwardRef />,
                    <p
                      className="truncate"
                    >
                      address
                    </p>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "address",
                      "className": "truncate",
                    },
                    "ref": null,
                    "rendered": "address",
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <ForwardRef />,
                    <p
                      className="truncate"
                    >
                      some description
                    </p>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "some description",
                      "className": "truncate",
                    },
                    "ref": null,
                    "rendered": "some description",
                    "type": "p",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                IN PROGRESS
              </p>,
              "className": "job-card-footer",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "IN PROGRESS",
              },
              "ref": null,
              "rendered": "IN PROGRESS",
              "type": "p",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
